core/modules/system/system.admin.inc

```diff
@@ -2242,7 +2235,12 @@ function system_run_cron() {
     backdrop_set_message(t('Cron run failed.'), 'error');
   }
 
-  backdrop_goto('admin/reports/status');
+  if ($_SERVER['HTTP_REFERER']) {
```

Comment:
Perhaps explore using `backdrop_get_destination()` here instead?

Reply:
I looked into that originally, but [it says](https://api.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_get_destination/1) "By default the current URL is returned." So that would only work if each and every place that calls this set a destination parameter in the URL. Currently none do, so I thought it easier to do it this way.